lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -341,19 +379,38 @@ public void setIdField(String idField) {
 
   @Override
   public void close() throws ConnectorException {
-    if (connection == null) {
-      // no-op
-      return;
-    }
-    try {
-      connection.close();
-    } catch (SQLException e) {
-      throw new ConnectorException("Error closing the connection", e);
+    for (Connection connection : connections) {
+      if (connection == null) {
+        // no-op
+        continue;
+      }
+      try {
+        connection.close();
+      } catch (SQLException e) {
+        throw new ConnectorException("Error closing the connection", e);

```

Comment:
If there are multiple connections and closing the first one causes an exception, we won't close the other connections because of the throw here.

Reply:
now we catch it and continue to try to close the rest.